### PR TITLE
fix: kill Receiver processes when long time passes with no messages

### PR DESCRIPTION
**Pull request by lemald**
#### Summary of changes
**Asana Ticket:** [💳 socket-proxy final fix: log socket activity and prune stale sockets](https://app.asana.com/0/584764604969369/1199174709467977/f)

@losvedir submitting for your review before I try putting it up on the server for real.

#### Reviewer Checklist
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Codecov)
- [ ] Meets ticket's acceptance criteria

### CODE REVIEW

File: lib/socket_proxy/receiver.ex [hunk 1]
```diff
@@ -4,7 +4,12 @@ defmodule SocketProxy.Receiver do
 
   @behaviour :ranch_protocol
 
-  defstruct socket: nil, destination_pids: [], port: nil
+  @staleness_check_interval_ms Application.compile_env(
```
Comment (losvedir) [inline lib/socket_proxy/receiver.ex:7]: Ooh, `compile_env`. I was going to call out using the `Application` environment in a module attribute, but actually it looks like that's okay now, using this macro. Huh.

**Comment by losvedir** [comment]
Looks good to me, if you want to test it live for a bit!

File: lib/socket_proxy/receiver.ex (hunk 1 above)
Comment (lemald) [reply lib/socket_proxy/receiver.ex:7]: I picked up the "`compile_env` in a module attribute" pattern from code that Paul wrote, so it must be good. 😅

**Review by losvedir — APPROVED** [review]
👍

**Pull request merged**